gossip/privdata/pvtdataprovider.go

```diff
@@ -314,8 +335,9 @@ func (pdp *PvtdataProvider) populateFromTransientStore(pvtdata rwsetByKeys, pvtd
 				continue
 			}
 			simRes := res.PvtSimulationResultsWithConfig
+			// simRes.PvtRwset will be nil if collection1 is retreived from transient store but current filter is for collection2
 			if simRes.PvtRwset == nil {
-				pdp.logger.Warningf("The PvtRwset of PvtSimulationResultsWithConfig for txID [%s] is nil. Skipping.", k.txID)
+				pdp.logger.Debugf("The PvtRwset of PvtSimulationResultsWithConfig for txID [%s] is nil. Skipping.", k.txID)
```

Comment:
Not related to this PR. We should make the `Next()` method in transient store to return the next matching entry instead of sending a nil `PvtRwSet`.

Reply:
I considered it, didn't want to touch the code that much though.